.hookz.yaml

```
@@ -1,16 +1,25 @@
-  version: 2.0
+  version: 2.1.1
   hooks:
     - type: pre-commit
       actions:
         - name: "Git Pull (Ensure there are no upstream changes that are not local)"
           exec: git
           args: ["pull"]
-        - name: "Go Tidy"
+        - name: "Go Tidy (Recursive)"
+          script: "
+            #!/bin/bash \n
+            echo -e Tidying all found go.mod occurrences\n
+            find . -name go.mod -print0 | xargs -0 -n1 dirname |  xargs -L 1 bash -c 'cd \"$0\" && pwd && go mod tidy' \n
+            "
+        - name: "Go Build (Ensure pulled modules do not break the build)"
           exec: go
-          args: ["mod", "tidy"]
+          args: ["build"]
         - name: "Update all go dependencies to latest"
           exec: go
           args: ["get", "-u", "./..."]
+        - name: "Run gofmt to format the code"
+          exec: gofmt
+          args: ["-w", "*/*.go"]
```

```suggestion
          args: ["-s", "-w", "**/*.go"]
```
Do you want to add simplification and recursive glob as a test?